api/db/changelog/db.changelog-166-add-access-module-tables.xml

```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+  xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+  <changeSet author="yonghao" id="changelog-166-add-access-module-tables">
+    <createTable tableName="access_module">
+      <column name="access_module_id" type="bigint" autoIncrement="true">
+        <constraints primaryKey="true" nullable="false"/>
+      </column>
+      <column name="access_module_enum" type="VARCHAR(80)">
+        <constraints unique="true" nullable="false"/>
+      </column>
+      <column name="expirable" type="boolean">
+        <constraints nullable="false"/>
+      </column>
+    </createTable>
+    <createTable tableName="access_module_user">
+      <column name="access_module_user_id" type="bigint" autoIncrement="true">
+        <constraints primaryKey="true" nullable="false"/>
+      </column>
+      <column name="access_module_id" type="bigint">
+        <constraints nullable="false" foreignKeyName="fk_acu_access_module_id" references="access_module(access_module_id)"/>
+      </column>
+      <column name="user_id" type="bigint">
+        <constraints nullable="false" foreignKeyName="fk_acu_user_id" references="user(user_id)"/>
+      </column>
+      <column name="completion_time" type="datetime">
+        <constraints nullable="false"/>
+      </column>
+      <column name="bypass_time" type="datetime">
+        <constraints nullable="true"/>
+      </column>
+      <column name="expire_time" type="datetime">
```

Comment:
I am not sure that we want this column, because it is derived from other columns.  If this somehow gets out of sync with the other columns, what is the correct behavior, and which do we trust as the source of truth?

The *information* in this column is certainly useful but I'd strongly prefer providing it in another way.  Perhaps as a readonly DB view, or a transient method in the Java entity object?

Would you mind removing it from this PR?  If we do ultimately decide it does in fact belong here, we can add it to the table then.

Reply:
Good point.  To clarify, this is derived from logic + config, not other columns. Initially I want this to be the new source of truth, but now I think you are correct. Now I see a problem of having this: We set expireDays to 3650 from 365, that requires update values in this column which is a really bad idea. I am removing this now.